props/saber_fett263_buttons.h

```diff
@@ -293,248 +660,3355 @@ class SaberFett263Buttons : public virtual PropBase {
+          break;
+        }
+        PlayMenuSound("maffirm.wav");
+        break;
+      case MENU_EXIT:
+        PlayMenuSound("mselect.wav");
+        MenuExit();
+        break;
+  #endif
+      default:
+        break;  
+      }
+  }
+
+// Edit Mode Dial
+  void MenuDial(int direction) {
+    switch (menu_type_) {
+      default:
+        break;
+      case MENU_PRESET:
+        if (direction > 0) {
+          if (SaberBase::IsOn()) {
+            next_preset_fast();
+          } else {
+            next_preset();
+          }
+        } else {
+          if (SaberBase::IsOn()) {
+            previous_preset_fast();
+          } else {
+            previous_preset();
+          }
+        }
+        break;
+      case MENU_VOLUME:
+        if (direction > 0) {
+          VolumeUp();
+        } else {
+          VolumeDown();
+        }
+        break;
+      case TRACK_PLAYER:
+        if (track_player_) {
+          track_player_->Stop();
+          track_player_.Free();
+        }
+        track_num_ += direction;
+        if (track_num_ > num_tracks_ - 1) track_num_ = 0;
+        if (track_num_ < 0) track_num_ = num_tracks_ - 1;
+        char playtrack[128];
+        RunCommandAndGetSingleLine("list_current_tracks", nullptr, track_num_, playtrack, sizeof(playtrack));
+  #ifdef ENABLE_AUDIO
+        MountSDCard();
+        EnableAmplifier();
+        track_player_ = GetFreeWavPlayer();
+        if (track_player_) {
+          track_player_->Play(playtrack);
+        } else {
+          STDOUT.println("No available WAV players.");
+        }
+  #else
+        STDOUT.println("Audio disabled.");
+  #endif
+        break;
+  #ifdef FETT263_EDIT_MODE_MENU
+      case MENU_TOP:
+        menu_top_pos_ += direction;
+        if (menu_top_pos_ <= 0) menu_top_pos_ = SUBMENUS;
+        if (menu_top_pos_ > SUBMENUS) menu_top_pos_ = 1;
+        switch (menu_top_pos_) {
+          case 1:
+            PlayMenuSound("mstyle.wav");
+            break;
+          case 2:
+            PlayMenuSound("mcolor.wav");
+            break;
+          case 3:
+            PlayMenuSound("mfont.wav");
+            break;
+          case 4:
+            PlayMenuSound("mtrack.wav");
+            break;
+          case 5:
+            PlayMenuSound("msetting.wav");
+            break;
+          case 6:
+            PlayMenuSound("mcopy.wav");
+            break;
+          case 7:
+            PlayMenuSound("mdelete.wav");
+            break;
+          case 8:
+            PlayMenuSound("mexit.wav");
+            break;
+       }
+       break;
+  #if NUM_BLADES > 1
+      case MENU_BLADE_STYLE:
+      case MENU_BLADE_COLOR:
+      case MENU_BLADE_SETTING:
+      case MENU_BLADE_COPY:            
+      case MENU_BLADE_LENGTH:
+        blade_num_ += direction;
+        if (blade_num_ > NUM_BLADES) blade_num_ = 1;
+        if (blade_num_ < 1) blade_num_ = NUM_BLADES;
+        BladePreview(blade_num_);
+        SFX_mnum.Select(blade_num_ - 1); 
```

Comment:
It would make more sense if blade_num_ was 0 ... NUM_BLADES-1 instead of 1...NUM_BLADES IMHO.

Reply:
updated

Reply:
Doesn't seem updated.

Reply:
Maybe I misunderstood the change, so if you turn the dial right (+1)  and it is greater than NUM_BLADES it circles around back to blade #1 - or - if you turn the dial left (-1) and it is less than 1 then you want last blade (which is equal to NUM_BLADES) to circle around in that direction. If we change to NUM_BLADES - 1 it would miss the last blade since blades are numbered 1 to NUM_BLADES instead of 0 to NUM_BLADES - 1.